protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1456,109 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	// The type of the object. The value MUST be one of "string", "number", "integer", "boolean", or "array"
+	// See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#headerObject
+	// Note: currently not implementing array as we are only implementing this in the operation response context
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string) error {
+	switch headerType {
+	case
+		"string":
+		if !(isQuotedString(defaultValue)) {
```

Comment:
Nit; you can remove the extra brackets here
```suggestion
		if !isQuotedString(defaultValue) {
```

Reply:
What I get for jumping back and forth between php and go 😞